### ISSUE TITLE
accessing file contents

### ISSUE THREAD

**Issue by JOJ0** [alexishevia/rn-dropbox-filepicker#4]
Hi,
I'm afraid to ask a stupid question again. I managed to integrate your component into my app. I can pick a file from dropbox. I can access the path to it via this.state.filepath, alright, but how would I access the contents of the file? A kind of file-handle or something. Again: Sorry to bother your with this, it's not really an issue with the component, I know....

In my case the picked file is an sqlite file. I have the components to access a local database file finished already. Actually what I really would like to do, is download the sqlite file to local storage, work with it, then upload again. I am not sure if your component is supposed to work like this. You state "file is connected and syncing to file ..." when the file was picked, so again: how would I access the contents of the file. I could also imagine adapting my app to directly edit the sqlite _on_ dropbox and don't have to download it first.

Some experiments and wild guessing shows that the file definitely was not "magically" downloaded to the local storage. There still is my sql directory and my locally stored database only:
```
state dropbox filepath: /discodos/discobase.db_2020-04-13_150912
dirContents Array [
  "SQLite",
]
dirSqlContents Array [
  "discobase.db",
]
```

I tried to read the packages code but sometimes these react js things and javascript in general do not make sense to me yet (I'm coming from python ;-)

maybe you could shed some light again! thanks a lot!
Jojo

**Comment by alexishevia** [comment]
Hey @JOJ0. Once you reach that `"file is connected and syncing to file ..."` message, it means your app now has both:
- A Dropbox access token
- A path to a Dropbox file

That was whole purpose of this package.

The "syncing to file" part of the message is very specific to the app that I was building when I created this package, so it's a little misleading if your app doesn't actually do any syncing.

To clarify: the "sync" portion is not included in this package.

Dropbox offers a Javascript SDK that you can use: https://www.npmjs.com/package/dropbox

I'm actually using that SDK to populate the filepicker in this repo. See: https://github.com/alexishevia/rn-dropbox-filepicker/blob/master/lib/api/loadDir.js#L1

You can probably find all the info you need to write your sync function in the Dropbox's SDK documentation: https://dropbox.github.io/dropbox-sdk-js/ 

You can also take a look at this other repo I have for reference on how to read/write to a Dropbox file: 
https://github.com/alexishevia/redux-file-sync/tree/master/src/dropbox/api
(just keep in mind I wrote that code specifically for my app's requirements, which may or may not match your app's requirements)

**Comment by JOJ0** [comment]
Hi Alexis,
thanks a lot for elaborating. That made everything clear. Especially thanks for pointing me to your redux-file-sync repo, I could already steal and adapt some code to my needs! ;-)

I am thinking about making the informational output "syncing to file ..." configurable in the package to make the component even better reusable. I am currently not quite sure how I will achieve this but will send you a pull request if I figure it out.

thanks so much again! all very helpful!

**Comment by alexishevia** [comment]
No problem.

I would definitely accept a PR to make messages configurable.

At the moment I have 3 hardcoded messages:
- LoggedOut: https://github.com/alexishevia/rn-dropbox-filepicker/blob/master/lib/LoggedIn.js#L11-L14
- LoggedIn: https://github.com/alexishevia/rn-dropbox-filepicker/blob/master/lib/LoggedIn.js#L11-L14
- FileSelected: https://github.com/alexishevia/rn-dropbox-filepicker/blob/master/lib/FileSelected.js#L11-L12

I would imagine accepting 3 optional props to override the default messages, something like:
```
<FilePicker
   appKey={APP_KEY}
   ... all the other props here ...
   loggedOutMsg={<Text>Your own message.</Text>}
   loggedInMsg={<Text>something else.</Text>}
   fileSelectedMsg={<Text>foo</Text>}
/>
```

When rendering the component we would check if the user sent one of these override props and use it, otherwise fallback to the default hardcoded messages.

**Comment by JOJ0** [comment]
Finally the PR: #5

**Comment by alexishevia** [comment]
Thanks for the PR!

I'm going to merge it now, and will try to generate a new version of this package on npm.
(It's been a while since the last time I created a new version, so it might take some time)

I'll let you know.

**Comment by alexishevia** [comment]
I published version 0.0.5 of this package to npm, which includes PR #5 

see: 
- [rn-dropbox-filepicker v0.0.5 @ npmjs](https://www.npmjs.com/package/rn-dropbox-filepicker/v/0.0.5)
- [changelog](https://github.com/alexishevia/rn-dropbox-filepicker/compare/0.0.4...0.0.5)

Thanks for the help!
